apps/omg_watcher/test/integration/invalid_exit_test.exs

```diff
@@ -0,0 +1,159 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.Watcher.Integration.InvalidExitTest do
+  use ExUnitFixtures
+  use ExUnit.Case, async: false
+  use OMG.API.Fixtures
+  use OMG.API.Integration.Fixtures
+  use Plug.Test
+  use Phoenix.ChannelTest
+
+  alias OMG.API
+  alias OMG.Eth
+  alias OMG.JSONRPC.Client
+  alias OMG.Watcher.Eventer.Event
+  alias OMG.Watcher.Integration.TestHelper, as: IntegrationTest
+  alias OMG.Watcher.Web.Channel
+
+  @moduletag :integration
+
+  @timeout 40_000
+  @eth API.Crypto.zero_address()
+
+  @endpoint OMG.Watcher.Web.Endpoint
+
+  @tag fixtures: [:watcher_sandbox, :stable_alice, :child_chain, :token, :stable_alice_deposits]
+  test "transaction which is using already spent utxo from exit and happened before end of m_sv causes to emit invalid_exit event ",
+       %{stable_alice: alice, stable_alice_deposits: {deposit_blknum, _}} do
+    # TODO remove this tx , use directly deposit_blknum to get_exit_data
+    tx = API.TestHelper.create_encoded([{deposit_blknum, 0, 0, alice}], @eth, [{alice, 10}])
+    {:ok, %{blknum: deposit_blknum}} = Client.call(:submit, %{transaction: tx})
+
+    bad_tx = API.TestHelper.create_recovered([{deposit_blknum, 0, 0, alice}], @eth, [{alice, 10}])
+
+    %{hash: bad_block_hash, number: bad_block_number, transactions: _} =
+      bad_block = API.Block.hashed_txs_at([bad_tx], 27_000)
+
+    {:module, BadChildChainBLock, _, _} = OMG.Watcher.Integration.BadChildChainBLock.create_module(bad_block)
+
+    JSONRPC2.Servers.HTTP.http(BadChildChainBLock, port: BadChildChainBLock.port())
+
+    {:ok, _, _socket} = subscribe_and_join(socket(), Channel.Byzantine, "byzantine")
+
+    {:ok, _} =
+      OMG.Eth.RootChain.submit_block(
+        bad_block_hash,
+        27,
+        1
+      )
+
+    IntegrationTest.wait_until_block_getter_fetches_block(deposit_blknum, @timeout)
+
+    %{
+      "txbytes" => txbytes,
+      "proof" => proof,
+      "sigs" => sigs,
+      "utxo_pos" => utxo_pos
+    } = IntegrationTest.get_exit_data(deposit_blknum, 0, 0)
+
+    {:ok, txhash} =
+      Eth.RootChain.start_exit(
+        utxo_pos,
+        txbytes,
+        proof,
+        sigs,
+        alice.addr
+      )
+
+    {:ok, %{"status" => "0x1"}} = Eth.WaitFor.eth_receipt(txhash, @timeout)
+    {:ok, current_child_block} = Eth.RootChain.get_current_child_block()
+
+    Application.put_env(
+      :omg_jsonrpc,
+      :child_chain_url,
+      "http://localhost:" <> Integer.to_string(BadChildChainBLock.port())
+    )
+
+    slow_exit_validator_block_margin =
+      Application.get_env(:omg_watcher, :slow_exit_validator_block_margin) *
```

Comment:
make sure that this `slow_exit_validator_block_margin` is properly and consistently documented in - `config.exs` where it is a default, in comments, in `ExitValidator` module and in the `.md` file in `docs`.

It should explain the purpose (that it is dependent on the SLA between child chain and users).

Reply:
I changed var `slow_exit_validator_block_margin ` to `margin_slow_validator` in the whole codebase to be consist with `M_SV`